### ISSUE TITLE
URL arguments that include encrypted `survey_url`, `verification_key` (which forces a verification wrapper on the survey, e.g., `WorkerId`), as well as arbitrary parameters that can be passed in and appear in the data and perhaps some other options that we think are important e.g., expiry.

### ISSUE THREAD

**Issue by markwhiting** [Watts-Lab/surveyor#5]
Basically we want our url to accept a few arguments that chooses which survey to show. Unencrypted it might look like: 

`surveyor.com/raw.githubusercontent.com/Watts-Lab/surveyor/main/surveys/CRT.csv?WorkerId=A123`

So we load the survey form that url and then we add the variable `WorkerId` with the value `A123` to the response we get from that URL (this should deal with arbitrary added variables of course).

However, the encrypted version might look like: 
`surveyor.com/wer234et34t5erg/ortrepoit234349i`

So with the encrypted one someone can't make guesses about how changing the URL would impact what happens.

The key for encoding the url arguments should be a static secret for now, e.g., `watts-lab` (and we can make it better as a future step)

**Comment by markwhiting** [comment]
@KaranSampath can you start working on this?

Basically we want our url to accept a few arguments that chooses which survey to show. Unencrypted it might look like: 

`surveyor.com/raw.githubusercontent.com/Watts-Lab/surveyor/main/surveys/CRT.csv?WorkerId=A123`

So we load the survey form that url and then we add the variable `WorkerId` with the value `A123` to the response we get from that URL (this should deal with arbitrary added variables of course).

However, the encrypted version might look like: 
`surveyor.com/wer234et34t5erg/ortrepoit234349i`

So with the encrypted one someone can't make guesses about how changing the URL would impact what happens.

The key for encoding the url arguments should be a static secret for now, e.g., `watts-lab` (and we can make it better as a future step)

**Comment by markwhiting** [comment]
Just adding a few notes after our meeting. Our steps for this are:
- [ ] rig URLs correctly, e.g., `https://surveyor.com/url=raw.githubusercontent.com/Watts-Lab/surveyor/main/surveys/CRT.csv&WorkerId=A123` with a format something like `https://surveyor.com/url={survey url}&[key=value for arbitrary extra variables]`
- [ ] an encryption approach for creating encrypted urls (e.g., encrypt endpoint) and for decrypting them too so they can be rendered. 
- [ ] joing these together with an approach something like the following: `https://surveyor.com/{secret url, and a secret string of key value pairs}/{researcher ID (so we can decode correctly)}`

**Comment by markwhiting** [comment]
I fixed the vanilla URL approach in #11.

@3hoursofsleep theres a few patterns there that might be useful to observe when creating the encrypted version. Notably, I had to use a `req.query` as opposed to a `req.params` because when using the parameters object I was running into issues where the slashes in the `survey_url` would be interpreted as slashes in the main URL. In any case, doing it with `req.query` seems to work fine. 

This should not be an issue in the encrypted version because we should be able to force the encrypted message to not have any escape characters in it.

@3hoursofsleep feel free to build on that PR when doing the next part.
